IPython/core/magics/display.py

```diff
@@ -59,12 +60,23 @@ def svg(self, line, cell):
         """Render the cell as an SVG literal"""
         display(SVG(cell))
 
+    @magic_arguments.magic_arguments()
+    @magic_arguments.argument(
+        '--isolated', action='store_true', default=False,
+        help="""Annotate the cell as 'isolated'.
+Isolated cells are rendered inside their own <iframe> tag"""
+    )
     @cell_magic
     def html(self, line, cell):
         """Render the cell as a block of HTML"""
-        display(HTML(cell))
-        
-    @cell_magic    
+        args = magic_arguments.parse_argstring(self.html, line)
+        html = HTML(cell)
+        if args.isolated:
+            display(html, metadata={'text/html':{'isolated':True}})
```

Comment:
We should make note of this metadata in http://jupyter.readthedocs.io/en/latest/reference/mimetype.html or wherever seems like a good choice. /cc @willingc